packages/core/components/NestedDropdown/tests/nestedDropdownHelpers.test.ts

```diff
@@ -21,32 +99,194 @@ describe('handleSearchTerm()', () => {
+            subgroupText: 'Tomato'
+          }
+        ]
+      ]
+    ]
     expect(result3).toEqual(expectedResult3)
     const result4 = handleSearchTerm('f', optsMemo)
     const expectedResult4: OptionsMemo = [
-      ['Fruits', ['Apple', 'Cantaloupe', 'Orange', 'Tomato']],
-      ['Fungi', ['Portobello', 'Morel', 'Chanterelle']]
+      [
+        'Fruits',
+        [
+          {
+            value: 'Fruits',
+            text: 'Fruits',
+            subgroupValue: 'Apple',
+            subgroupText: 'Apple'
+          },
+          {
+            value: 'Fruits',
+            text: 'Fruits',
+            subgroupValue: 'Cantaloupe',
+            subgroupText: 'Cantaloupe'
+          },
+          {
+            value: 'Fruits',
+            text: 'Fruits',
+            subgroupValue: 'Orange',
+            subgroupText: 'Orange'
+          },
+          {
+            value: 'Fruits',
+            text: 'Fruits',
+            subgroupValue: 'Tomato',
+            subgroupText: 'Tomato'
+          }
+        ]
+      ],
+      [
+        'Fungi',
+        [
+          {
+            value: 'Fungi',
+            text: 'Fungi',
+            subgroupValue: 'Portobello',
+            subgroupText: 'Portobello'
+          },
+          {
+            value: 'Fungi',
+            text: 'Fungi',
+            subgroupValue: 'Morel',
+            subgroupText: 'Morel'
+          },
+          {
+            value: 'Fungi',
+            text: 'Fungi',
+            subgroupValue: 'Chanterelle',
+            subgroupText: 'Chanterelle'
+          }
+        ]
+      ]
     ]
     expect(result4).toEqual(expectedResult4)
   })
   it('Filter out the subvalues', () => {
     const result = handleSearchTerm('ap', optsMemo)
-    const expectedResult: OptionsMemo = [['Fruits', ['Apple']]]
+    const expectedResult: OptionsMemo = [
+      [
+        'Fruits',
+        [
+          {
+            value: 'Fruits',
+            text: 'Fruits',
+            subgroupValue: 'Apple',
+            subgroupText: 'Apple'
+          }
+        ]
+      ]
+    ]
     expect(result).toEqual(expectedResult)
     const result2 = handleSearchTerm('ca', optsMemo)
     const expectedResult2: OptionsMemo = [
-      ['Fruits', ['Cantaloupe']],
-      ['Vegetables', ['Carrot']]
+      [
+        'Fruits',
+        [
+          {
+            value: 'Fruits',
+            text: 'Fruits',
+            subgroupValue: 'Cantaloupe',
+            subgroupText: 'Cantaloupe'
+          }
+        ]
+      ],
+      [
+        'Vegetables',
+        [
+          {
+            value: 'Vegetables',
+            text: 'Vegetables',
+            subgroupValue: 'Carrot',
+            subgroupText: 'Carrot'
+          }
+        ]
+      ]
     ]
     expect(result2).toEqual(expectedResult2)
     const result3 = handleSearchTerm('tomato', optsMemo)
     const expectedResult3: OptionsMemo = [
-      ['Fruits', ['Tomato']],
-      ['Vegetables', ['Tomato']]
+      [
```

Comment:
it seems like a better runtime data structure here would be a tuple. the text indexes could be optional. and here you have an ordered list of your nested dropdown options.

[  [value, text], [ [subValue, subText], [subValue, subText] ]  ]